stacks_queues/queue.py

```diff
@@ -15,47 +18,71 @@ def __init__(self):
         self.front = -1
         self.rear = -1
         self.size = 0
-      
 
     def enqueue(self, element):
         """ Adds an element to the Queue
             Raises a QueueFullException if all elements
             In the store are occupied
             returns None
         """
-        pass
+        if self.front == -1:
+            self.front = self.rear = 0
+
+        if self.full():
+            # raise QueueFullException("Queue is full")
+            buffer_increase = int(self.buffer_size * 0.5)
+            self.store = self.store[:self.rear] + [None] * buffer_increase + self.store[self.front:]
+            self.buffer_size += buffer_increase
+            self.front += buffer_increase
+
+        self.store[self.rear] = element
+        self.rear = (self.rear + 1) % self.buffer_size
+        self.size += 1
 
     def dequeue(self):
```

Comment:
👍

If the size returns to 0, consider resetting front and rear (setting them back to the initialized state).